### DISCUSSION TITLE
Renaming to Main

### DISCUSSION THREAD

**Discussion by polandll** [stargate/stargate#1489] [RFCs]
As of October 2020, all new source code repositories created on GitHub have a primary branch named "main" instead of "master" as part of the company's effort to remove unnecessary references to slavery and replace them with more inclusive terms. This “fix forward first” approach is the first step in making systemic changes to the naming conventions. 

Github is also in the process of renaming their own internal repositories. This reflects the difficulty that Github and other organizations face – it’s hard to rename something that is in active use.

Hard, but not impossible, and it’s the right thing to do. We at the Stargate project are following suit. We’ll use “main” as the name of our primary branch going forward, and we are actively renaming existing primary branches to be named “main”. The Docs repo is the first to get renamed. Others will follow. We can’t let inertia be an impediment. We want our work on Stargate to be open, transparent, and inclusive. This is a necessary step to align with the virtue of inclusivity.

**Comment by dougwettlaufer** [comment]
Once changed, anyone that has `master` checked out will need to run

```sh
git branch -m master main
git fetch origin
git branch -u origin/main main
git remote set-head origin -a
```

**Reply by jeffreyscarpenter** [reply]
I have an idea for a relatively seamless way to accomplish this transition. We have work in progress for Stargate v2 on the `v2.0.0` branch. When it comes time to make v2 the official version, I propose we just rename `v2.0.0` to `main` and rename `master` to `v1` or something like that. At this time we'd make any changes in downstream projects that need to work off of the "latest" version.  If we make this the official plan we can announce it now and give all users ample time for preparation.

Note that we have been merging all v1 changes from `master` into `v2.0.0`. Therefore I do not see there being value in merging `v2.0.0` back to `master`, since we have been doing PR reviews on merges to `v2.0.0` from the beginning of the v2 effort.

**Comment by tatu-at-datastax** [comment]
This renaming was completed as #1490, so that:

* Branch `v1` refers to former `master` (for Stargate V1)
* Branch `main` refers to former `v2.0.0` (for Stargate V2)
* Default branch is now `main`
